annotation-file-utilities/src/annotator/find/ASTPathCriterion.java

```diff
@@ -383,9 +383,8 @@ private Tree getNext(Tree actualNode, ASTPath astPath, int ix) {
                 } else if (astNode.childSelectorIs(ASTPath.BOUND)) {
                   return arg < 0 ? clazz.getExtendsClause()
                           : clazz.getImplementsClause().get(arg);
-                } else {
-                    return null;
                 }
+                return null;
```

Comment:
I see.  I had overlooked that lines 386 and 387 were deleted.  I would restore them, and then insert the return null between lines 382 and 383.  Then every clause has its own return statement, which eases reasoning.  You could still insert the throw statement to emphasize the lack of a fallthrough case.